modules/naplicense/src/licenseservice.cpp

```diff
@@ -51,26 +51,44 @@ namespace nap
 	}
 
 
-	static bool rsaVerifyFile(const std::string& publicKey, const std::string& licenseFile, const std::string& signatureFile)
+	static CryptoPP::PK_Verifier* getVerifier(const std::string& publicKey, nap::SigningScheme signingScheme)
+	{
+		CryptoPP::StringSource pub_file(publicKey.c_str(), true, new CryptoPP::HexDecoder);
+		CryptoPP::PK_Verifier* verifier = nullptr;
+		if (signingScheme == nap::SigningScheme::RSASS_PKCS1v15_SHA1)
+			verifier = new CryptoPP::RSASS<CryptoPP::PKCS1v15, CryptoPP::SHA1>::Verifier(pub_file);
+		else if (signingScheme == nap::SigningScheme::RSASS_PKCS1v15_SHA224)
+			verifier = new CryptoPP::RSASS<CryptoPP::PKCS1v15, CryptoPP::SHA224>::Verifier(pub_file);
+		else if (signingScheme == nap::SigningScheme::RSASS_PKCS1v15_SHA256)
+			verifier = new CryptoPP::RSASS<CryptoPP::PKCS1v15, CryptoPP::SHA256>::Verifier(pub_file);
+		else if (signingScheme == nap::SigningScheme::RSASS_PKCS1v15_SHA384)
+			verifier = new CryptoPP::RSASS<CryptoPP::PKCS1v15, CryptoPP::SHA384>::Verifier(pub_file);
+		else if (signingScheme == nap::SigningScheme::RSASS_PKCS1v15_SHA512)
+			verifier = new CryptoPP::RSASS<CryptoPP::PKCS1v15, CryptoPP::SHA512>::Verifier(pub_file);
+
+		return verifier;
```

Comment:
Can you return a `unique_ptr<T>` instead of raw pointer here? We don't allow the use of raw pointers this way, although limited to this scope it can be a potential memory leak. Maybe it's a good idea to change the name to `createVerifier`?